caput/mpiarray.py

```diff
@@ -730,39 +787,38 @@ def to_hdf5(
                 )
 
         mode = "a" if create else "r+"
-
         fh = misc.open_h5py_mpi(f, mode, self.comm)
 
-        start = self.local_offset[self.axis]
-        end = start + self.local_shape[self.axis]
-
-        # Construct slices for axis
-        sel = ([slice(None, None)] * self.axis) + [slice(start, end)]
-        sel = _expand_sel(sel, self.ndim)
-
         # Check that there are no null slices, otherwise we need to turn off
         # collective IO to work around an h5py issue (#965)
         no_null_slices = self.global_shape[self.axis] >= self.comm.size
 
-        # Split the axis to get the IO size under ~2GB (only if MPI-IO)
-        split_axis, partitions = self._partition_io(skip=(not fh.is_mpi))
-
         # Only use collective IO if:
         # - there are no null slices (h5py bug)
         # - we are not distributed over axis=0 as there is no advantage for
         #   collective IO which is usually slow
         # - unless we want to use compression/chunking
         # TODO: change if h5py bug fixed
+        # https://github.com/h5py/h5py/issues/965
         # TODO: better would be a test on contiguous IO size
         use_collective = (
             fh.is_mpi and no_null_slices and (self.axis > 0 or compression is not None)
```

Comment:
On second thought, we may want to include here that compression is turned off for `HDF5`. Because since we are not doing compression, there is a possibility of using collective IO.